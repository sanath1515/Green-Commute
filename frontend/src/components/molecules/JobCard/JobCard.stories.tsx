import React from "react";
import JobCard from "./JobCard";
import hp from "../../../assests/hp.webp";

export default {
  title: "Molecules/JobCard",
  component: JobCard,
};

const Template = (args) => <JobCard {...args} />;

export const Default = Template.bind({});

Default.args = {
  src: hp,
  days: 2,
  jobRole: "User Experience Designer",
  companyName: "Hp",
  location: "Hyderabad, Telanagana ,India",
  commuteRoute: ["bus", "car"],
};
