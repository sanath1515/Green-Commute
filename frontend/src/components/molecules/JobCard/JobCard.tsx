import React from "react";

import Icons1 from "../../atoms/Transporticons/icons";
import { COMMUTE_ROUTES } from "../../../Constants";
import { makeStyles } from "@material-ui/styles";
import {
  Card,
  CardActions,
  CardContent,
  CardHeader,
  IconButton,
  ListItem,
  Typography,
} from "@material-ui/core";
import MoreHorizIcon from "@material-ui/icons/MoreHoriz";
import { Avatars } from "../../atoms/Avatar/Avatar";

const useStyles = makeStyles({
  icon: {
    color: "#9bbdcb",
    paddingRight: "8px",
    marginTop: "6px",
  },
  card: {
    maxWidth: "330px",
    height: "308px",
    elevation: 2,
    border: "solid 1px #e3f3f6",
  },
  avatar: {
    width: "80px",
    height: "80px",
    marginLeft: "8px",
    marginTop: "6px",
  },
  list: {
    paddingRight: "10px",
  },
  role: {
    marginTop: "-7px",
    color: "#324552",
    marginLeft: "10px",
    paddingBottom: "8px",
  },
  company: {
    marginTop: "-4px",
    color: "#5f7381",
    marginLeft: "10px",
    marginBottom: "10px",
  },
  location: {
    marginTop: "-4px",
    color: "#5f7381",
    marginLeft: "10px",
    marginBottom: "18px",
  },
  routes: {
    color: "#19293b",
    marginLeft: "10px",
    paddingBottom: "15px",
  },

  days: {
    color: "#5f7381",
    marginBottom: "50px",
    textAlign: "right",
    marginTop: "6px",
  },
});

type CardProps = {
  src?: string;
  days?: number;
  jobRole?: string;
  companyName?: string;
  location?: string;
  commuteRoute?: Array<string>;
};

const JobCard = ({
  src,
  days,
  jobRole,
  companyName,
  location,
  commuteRoute,
}: CardProps) => {
  const classes = useStyles();
  return (
    <>
      <Card className={classes.card} elevation={0}>
        <CardHeader
          avatar={<Avatars src={src} className={classes.avatar} />}
          action={
            <IconButton>
              <MoreHorizIcon className={classes.icon} />
            </IconButton>
          }
          title={
            <Typography variant="body2" className={classes.days}>
              {days}d
            </Typography>
          }
        />
        <CardContent>
          <Typography variant="h6" className={classes.role}>
            {jobRole}
          </Typography>

          <Typography variant="body2" className={classes.company}>
            {companyName}
          </Typography>

          <Typography variant="body2" className={classes.location}>
            {location}
          </Typography>

          <Typography variant="caption" className={classes.routes}>
            {COMMUTE_ROUTES.text}
          </Typography>
          <br />

          <ListItem className={classes.list}>
            {commuteRoute!.map((item: string) => (
              <Icons1 name={item} className={classes.icon} />
            ))}
          </ListItem>
        </CardContent>
        <CardActions></CardActions>
      </Card>
    </>
  );
};

export default JobCard;
