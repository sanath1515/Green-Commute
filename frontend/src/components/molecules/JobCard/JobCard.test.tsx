/**
 * @jest-environment jsdom
 */
import { fireEvent, render, screen } from "@testing-library/react";
import JobCard from "./JobCard";

describe("Jobcard Component", () => {
  test("Matches Snapshot", () => {
    const { asFragment } = render(
      <JobCard
        src="./image.sng"
        days={2}
        jobRole="User Experience Designer"
        companyName="Hp"
        location="Hyderabad,Telanagana, India"
        commuteRoute={[]}
        key="1"
      />
    );
    expect(asFragment.name).toMatchSnapshot();
  });
});
