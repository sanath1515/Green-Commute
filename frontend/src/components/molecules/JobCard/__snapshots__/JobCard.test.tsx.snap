// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jobcard Component Matches Snapshot 1`] = `"asFragment"`;
